### PR TITLE
PI-3744 Prepare and map offender delta events before publishing notifications

### PR DESCRIPTION
**Pull request by serdaroguzhanakin**
## What

Refactored the offender_delta processing flow so that all mapping and domain event handling is completed **before** any notifications are published.

The batch is now prepared and removed from the database atomically, and only then are notifications published to SNS and Telemetry.

## Why

Previously, notifications were published inside the same transactional flow as delta processing.  
If a failure occurred after a notification was published, the transaction would roll back but the SNS message would already have been sent, leaving the offender_delta record undeleted and causing repeated publishes.

This change eliminates that risk by ensuring:
- All mapping and domain event handling happens up front
- offender_delta records are deleted only after successful preparation
- Notifications are published only after the transaction has completed successfully

This prevents the system from getting stuck in a loop repeatedly publishing the same events.

## Notes

- Behaviour and event semantics are unchanged
- The change is intentionally scoped to error handling and flow control
- Additional tests cover the updated prepare → publish lifecycle and edge cases

### CODE REVIEW

File: projects/offender-events-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/OffenderDeltaPoller.kt [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
 import org.springframework.scheduling.annotation.Scheduled
 import org.springframework.stereotype.Service
-import org.springframework.transaction.annotation.Transactional
 import uk.gov.justice.digital.hmpps.integrations.delius.offender.OffenderDeltaService
 
 @Service
 class OffenderDeltaPoller(private val service: OffenderDeltaService) {
-    @Transactional
     @Scheduled(fixedDelayString = "\${offender-events.fixed-delay:100}")
+    @WithSpan("POLL offender_delta", kind = SpanKind.SERVER)
```
[the remark this answers is not in the document]
Comment (serdaroguzhanakin) [reply projects/offender-events-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/OffenderDeltaPoller.kt]: done
